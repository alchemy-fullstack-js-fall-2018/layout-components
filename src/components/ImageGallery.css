.container {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
    grid-gap: 1rem;
    padding: 1rem;
    img{
      width: 100%;
    }
}