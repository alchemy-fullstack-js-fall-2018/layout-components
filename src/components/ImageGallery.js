import Component from './Component';
import styles from './ImageGallery.css';

export default class ImageGallery extends Component {
  render() {
    const images = this.props.images.map(({ image }) => {
      return `<img src="${image}">`;
    });

    return `
      <section>
        <div class="${styles.container}">
          ${images.join(' ')}
        </div>
      </section>
    `;
  }
}
