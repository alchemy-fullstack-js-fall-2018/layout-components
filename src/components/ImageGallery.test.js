import ImageGallery from './ImageGallery';

describe('Image Gallery Component', () => {
  it('is an image gallery', () => {
    const images = [
      {
        image: ''
      }
    ];

    const imageGallery = new ImageGallery({ images });

    expect(imageGallery.render).toMatchSnapshot();
  });
});
