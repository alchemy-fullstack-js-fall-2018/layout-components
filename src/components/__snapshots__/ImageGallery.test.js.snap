// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery Component is an image gallery 1`] = `[Function]`;
